src/analysis/convex/function.lean

```diff
@@ -261,6 +307,29 @@ lemma linear_order.concave_on_of_lt (hs : convex 𝕜 s)
      a • f x + b • f y ≤ f (a • x + b • y)) : concave_on 𝕜 s f :=
 @linear_order.convex_on_of_lt _ _ (order_dual β) _ _ _ _ _ _ s f hs hf
 
+/-- For a function on a convex set in a linear ordered space (where the order and the algebraic
+structures aren't necessarily compatible), in order to prove that it is convex, it suffices to
+verify the inequality `f (a • x + b • y) ≤ a • f x + b • f y` for `x < y` and positive `a`, `b`. The
+main use case is `E = 𝕜` however one can apply it, e.g., to `𝕜^n` with lexicographic order. -/
+lemma linear_order.strict_convex_on_of_lt (hs : convex 𝕜 s)
+  (hf : ∀ ⦃x y : E⦄, x ∈ s → y ∈ s → x < y → ∀ ⦃a b : 𝕜⦄, 0 < a → 0 < b → a + b = 1 →
+    f (a • x + b • y) < a • f x + b • f y) : strict_convex_on 𝕜 s f :=
+begin
+  refine ⟨hs, λ x y hx hy hxy a b ha hb hab, _⟩,
+  wlog h : x ≤ y using [x y a b, y x b a],
+  { exact le_total _ _ },
+  exact hf hx hy (h.lt_of_ne hxy) ha hb hab,
+end
+
+/-- For a function on a convex set in a linear ordered space (where the order and the algebraic
+structures aren't necessarily compatible), in order to prove that it is concave it suffices to
+verify the inequality `a • f x + b • f y ≤ f (a • x + b • y)` for `x < y` and positive `a`, `b`. The
+main use case is `E = ℝ` however one can apply it, e.g., to `ℝ^n` with lexicographic order. -/
```

Comment:
```suggestion
main use case is `E = 𝕜` however one can apply it, e.g., to `𝕜^n` with lexicographic order. -/
```

Reply:
Actually I left that one on purpose because the point seemed to give an actual real life people-actually-care-about-it concrete example. What do you think?

Reply:
I salute your good intentions but I think I'd still go with `𝕜` especially in view of the corresponding comment above for `linear_order.strict_convex_on_of_lt`.